packages/edit-post/src/components/sidebar/post-template/index.js

```diff
@@ -64,19 +65,7 @@ export default function PostTemplate() {
 
 function PostTemplateToggle( { isOpen, onClick } ) {
 	const templateTitle = useSelect( ( select ) => {
-		const templateSlug =
-			select( editorStore ).getEditedPostAttribute( 'template' );
-
-		const settings = select( editorStore ).getEditorSettings();
-		if ( settings.availableTemplates[ templateSlug ] ) {
-			return settings.availableTemplates[ templateSlug ];
-		}
-
-		const template = select( coreStore )
-			.getEntityRecords( 'postType', 'wp_template', { per_page: -1 } )
-			?.find( ( { slug } ) => slug === templateSlug );
-
-		return template?.title.rendered;
+		return select( editPostStore ).getEditedPostTemplate()?.title;
```

Comment:
Should we fall back to the slug like the old panel used to?

```js
const template = select( editPostStore ).getEditedPostTemplate();
return template?.title ?? template?.slug;
```

Genuine question. I'm not sure what circumstances would lead to an undefined `title`.